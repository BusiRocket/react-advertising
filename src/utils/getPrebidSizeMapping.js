export default (sizeMappingName, sizeMappings) => {
    if (sizeMappingName) {
        const sizeMapping = sizeMappings && sizeMappings[sizeMappingName];

        if (sizeMapping) {
            return sizeMapping
                .map(mapping => {
                    const { viewPortSize: [minWidth], sizes } = mapping;
                    return { minWidth, sizes };
                })
                .sort((m1, m2) => m2.minWidth - m1.minWidth);
        }
    }
    return null;
};
