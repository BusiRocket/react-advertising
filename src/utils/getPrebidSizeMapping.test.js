import getPrebidSizeMapping from './getPrebidSizeMapping';

const correctSizeMappingName = 'foo';
const incorrectSizeMappingName = 'bar';
const sizeMappings = {
    foo: [
        {
            viewPortSize: [200, 300],
            sizes: 'sizes'
        },
        {
            viewPortSize: [100],
            sizes: 'sizes'
        }
    ]
};

describe('When I get a Prebid size mapping', () => {
    describe('and size mapping name is undefined', () => {
        let result;
        beforeEach(() => (result = getPrebidSizeMapping()));
        describe('the result', () => it('is null', () => expect(result).toBeNull()));
    });
    describe('with a size mapping name', () => {
        describe('and no size mappings', () => {
            let result;
            beforeEach(() => (result = getPrebidSizeMapping(correctSizeMappingName)));
            describe('the result', () => it('is null', () => expect(result).toBeNull()));
        });
        describe('and size mappings that contain the mapping with the name', () => {
            let result;
            beforeEach(() => (result = getPrebidSizeMapping(correctSizeMappingName, sizeMappings)));
            describe('the result', () => it('is correct', () => expect(result).toMatchSnapshot()));
        });
        describe('and size mappings that do not contain the mapping with the name', () => {
            let result;
            beforeEach(() => (result = getPrebidSizeMapping(incorrectSizeMappingName, sizeMappings)));
            describe('the result', () => it('is null', () => expect(result).toBeNull()));
        });
    });
});
