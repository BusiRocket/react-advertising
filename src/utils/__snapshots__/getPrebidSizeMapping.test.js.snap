// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I get a Prebid size mapping with a size mapping name and size mappings that contain the mapping with the name the result is correct 1`] = `
Array [
  Object {
    "minWidth": 200,
    "sizes": "sizes",
  },
  Object {
    "minWidth": 100,
    "sizes": "sizes",
  },
]
`;
